cmd/console/deploy/status.go

```diff
@@ -0,0 +1,82 @@
+package deploy
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/mia-platform/miactl/factory"
+	"github.com/mia-platform/miactl/sdk"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func NewStatusCmd() *cobra.Command {
+	var (
+		baseURL     string
+		apiToken    string
+		projectId   string
+		environment string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "status",
+		Short: "verify status of deploy pipeline",
+		Args: func(cmd *cobra.Command, args []string) error {
+			return cobra.ExactArgs(1)(cmd, args)
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			baseURL = viper.GetString("apibaseurl")
+			apiToken = viper.GetString("apitoken")
+			projectId = viper.GetString("project")
+
+			if baseURL == "" {
+				return errors.New("API base URL not specified nor configured")
+			}
+			if apiToken == "" {
+				return errors.New("missing API token - please login")
+			}
+			if projectId == "" {
+				return cmd.MarkFlagRequired("project")
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			f, err := factory.FromContext(cmd.Context(), sdk.Options{
+				APIBaseURL: baseURL,
+				APIToken:   apiToken,
+			})
+			if err != nil {
+				return err
+			}
+
+			pipelineId, err := strconv.Atoi(args[0])
+			if err != nil {
+				f.Renderer.Error(fmt.Errorf("unexpected pipeline id: %w", err)).Render()
+				return nil
+			}
+
+			statusResponse, err := f.MiaClient.Deploy.GetDeployStatus(projectId, pipelineId, environment)
```

Comment:
What happen if deploy fails? 

Using it in a CD environment, how it is possible to know if status is not successful?

Reply:
At the moment, the command just visualize pipeline status, whichever it is.

The command can be changed, so that it could return an error in case of `failed` or `canceled` pipeline.
However, to run this command in a CD environment it is necessary to first extract the pipeline id. This may be achieved by parsing the output of the corresponding trigger deploy pipeline command.